File: 0167-two-sum-ii-input-array-is-sorted/0167-two-sum-ii-input-array-is-sorted.py
class Solution(object):
    def twoSum(self, numbers, target):
        """
        :type numbers: List[int]
        :type target: int
        :rtype: List[int]
        """
        left_index = 1
        right_index = len(numbers)
        while (left_index < right_index):
            left_num = numbers[left_index - 1]
            right_num = numbers[right_index - 1]
            sum_left_right = left_num + right_num
            if sum_left_right == target:
                return [left_index, right_index]
            elif sum_left_right < target:
                left_index += 1
            else:
                right_index -= 1
        

File: 0167-two-sum-ii-input-array-is-sorted/0167-two-sum-ii-input-array-is-sorted.java
class Solution {
    public int[] twoSum(int[] numbers, int target) {
        int low = 0;
        int high = numbers.length - 1;

        while (low < high) {
            int curr = numbers[low] + numbers[high];

            if (curr == target) {
                return new int[]{low + 1, high + 1};
            } else if (curr < target) {
                low++;
            } else {
                high--;
            }
        }

        return new int[]{-1, -1};
    }
}